libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -992,6 +1018,10 @@ bool AC_WPNav::set_spline_origin_and_destination(const Vector3f& origin, const V
     _flags.segment_type = SEGMENT_SPLINE;
     _flags.new_wp_destination = true;   // flag new waypoint so we can freeze the pos controller's feed forward and smooth the transition
 
+    // initialise yaw related variables
+    _track_length_xy = safe_sqrt(sq(_destination.x - _origin.x)+sq(_destination.y - _origin.y));  // horizontal track length (used to decide if we should update yaw)
+    _flags.wp_yaw_set = false;
```

Comment:
Humm, interesting, by looking at code I would say it was needed here and also in the straight waypoint. How does the yaw get unset when changing from one waypoint to the other? From what I could see the `wp_and_spline_init` method is only called when entering Auto, not when changing waypoint.